### PR TITLE
[go_router] Factor out test helpers from go_router_test

### PR DESCRIPTION
**Pull request by guidezpl**
This is to avoid having to import the library from a test file in `error_scheen_helpers.dart`.

Remove a TODO as well.

No CHANGELOG change: test-only change

### CODE REVIEW

File: packages/go_router/pubspec.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 name: go_router
 description: A declarative router for Flutter based on Navigation 2 supporting
   deep linking, data-driven routes and more
-version: 4.1.0
+version: 4.1.1
```
Comment (stuartmorgan) [inline packages/go_router/pubspec.yaml:4]: Test-only changes shouldn't change the version.

**Review by stuartmorgan — CHANGES_REQUESTED** [review]

**Comment by stuartmorgan** [comment]
FYI it's fine to take the "No CHANGELOG change:" PR description annotation approach in this PR (rather than doing a `## NEXT` CHANGELOG entry).

**Review by johnpryan — APPROVED** [review]
Thanks!

**Comment by guidezpl** [comment]
Do I have to do something for the publishable check to update?

**Comment by johnpryan** [comment]
IIRC it will pass when it's merged into `main`

**Comment by stuartmorgan** [comment]
> Do I have to do something for the publishable check to update?

Usually pushing a new commit does it; I'm not sure why it didn't here. Maybe empty commits don't work? (Unfortunately I don't know where the caching of the PR description our script receives is happening or why; it's not part of our pipeline.)

I would try either a merge or rebase from `main`, or an amend commit that changes the commit message slightly combined with a force-push.

**Comment by stuartmorgan** [comment]
> IIRC it will pass when it's merged into `main`

Checks with failing presubmits shouldn't be merged to `main`.

**Comment by stuartmorgan** [comment]
I have no idea why it's still failing; it definitely shouldn't be. Feel free to land manually, and I'll file an issue to try to repro this with instrumented tooling.

**Comment by guidezpl** [comment]
I think you need to approve first 
image

**Pull request merged**